test/extended/etcd/leader_changes.go

```diff
@@ -16,11 +16,15 @@ var _ = g.Describe("[sig-etcd] etcd", func() {
 	defer g.GinkgoRecover()
 	oc := exutil.NewCLI("etcd-leader-change").AsAdmin()
 	g.It("leader changes are not excessive", func() {
+		t := g.GinkgoT()
 		prometheus, err := client.NewE2EPrometheusRouterClient(oc)
 		o.Expect(err).ToNot(o.HaveOccurred())
 		g.By("Examining the rate of increase in the number of etcd leadership changes for last five minutes")
 		result, _, err := prometheus.Query(context.Background(), "increase((max by (job) (etcd_server_leader_changes_seen_total) or 0*absent(etcd_server_leader_changes_seen_total))[15m:1m])", time.Now())
 		o.Expect(err).ToNot(o.HaveOccurred())
-		o.Expect(result.(model.Vector)[0].Value).To(o.BeNumerically("==", 0))
+		leaderElectionLastFiveMinutes := result.(model.Vector)[0].Value
+		if leaderElectionLastFiveMinutes != 0 {
+			t.Errorf("Leader changes observed last 5m %q: Leader changes are a result of stopping the etcd leader process or from latency (disk or network), review etcd performance metrics", leaderElectionLastFiveMinutes)
```

Comment:
`o.Expect(fmt.Errorf("***your stuff here***).ToNot(o.HaveOccurred())`  it gives you a line number that way

Reply:
cool thx